xbmc/cores/AudioEngine/Sinks/AESinkStarfish.cpp

```diff
@@ -79,17 +92,14 @@ CAESinkStarfish::~CAESinkStarfish() = default;
 
 bool CAESinkStarfish::Initialize(AEAudioFormat& format, std::string& device)
 {
-  m_format = format;
```

Comment:
One point here:
We basically do the initial copy to "local m_format" at first. Reason is: We might hit a non supported format where we have to error out. That way the "format reference" passed in is only changed at the very end, when we succeed with all our setup / work, while you overwrite the reference passed by AE now directly in the first place.

What do you think?

Reply:
Sure sounds good. The reasoning behind this change was that previously it was kind of a mixture between using the global `m_format` and `format` so I made it at least consistant. It's probably better to work on a copy first then apply that copy to the reference and the global `m_format` so we don't end up using non negotiated formats anywhere

Reply:
Changed the code to take a copy as previously and made sure only the copy was used for setting up the sink then copying it back in the end.